packages/protocol-kit/src/SafeProvider.ts

```diff
@@ -56,14 +65,20 @@ class SafeProvider {
   }
 
   async getExternalSigner(): Promise<AbstractSigner | undefined> {
-    // If the signer is not an Ethereum address, it should be a private key
-    if (this.signer && !ethers.isAddress(this.signer)) {
-      const privateKeySigner = new ethers.Wallet(this.signer, this.#externalProvider)
-      return privateKeySigner
-    }
-
-    if (this.signer) {
-      return this.#externalProvider.getSigner(this.signer)
+    if (typeof this.signer === 'string') {
+      // If the signer is not an Ethereum address, it should be a private key
+      if (this.signer && !ethers.isAddress(this.signer)) {
```

Comment:
I think if it reached this point, we can assume that `this.signer` is defined. Please, remove the left-hand side conditionhere and below.

Reply:
done!